src/lib/internal.ts

```diff
@@ -4,12 +4,21 @@ import { NoActiveEntityError } from './errors';
 import { EntityStateModel } from './models';
 
 /**
- * type alias for javascript object literal
+ * Type alias for an object literal.
+ * Only allows strings as keys.
  */
-export interface HashMap<T> {
-  [id: string]: T;
+export interface Dictionary<T> {
+  [key: string]: T;
 }
 
+/**
+ * Type alias for an object literal.
+ * Only allows strings as keys.
+ * @deprecated Use Dictionary type instead
+ * @see Dictionary
+ */
+export type HashMap<T> = Dictionary<T>;
```

Comment:
I left it there in case some of the users was already using that type but I overlooked that it was in the `internal` file anyways. Thanks for the heads-up